File: probability/coinflips.cpp
#include <iostream>
//#include <iomanip>
#include <cstdlib>
#include <ctime>
#include <math.h>
//#include <vector>
// #include "mymath.cpp" // Tau not pi

int main()
{

  int numTrials = 100000;
	int heads = 0;
	int tails = 0;
	float pHeads = 0.5;

  std::srand(std::time(0));
	for (int i = 1; i <= numTrials; i++) {
		float randomNumber = std::rand()/( static_cast< float >( RAND_MAX ) + 1.f );
		if (randomNumber < pHeads) {
			heads++;
		} else {
			tails++;
		}
	}
	std::cout << "In " << numTrials << " trials there were " << heads << " heads and " << tails << " tails\n";
	std::cout << "ERCENT HEADS: " <<
    100.f * static_cast< float >(heads) / static_cast< float >(numTrials) << " percent";

}

File: probability/coinflips.go
package main

import (
	"fmt"
	"math/rand"
)

func main() {
	var numTrials = 100000
	var heads = 0
	var tails = 0
	pHeads := 0.5

	rd := rand.New(rand.NewSource(99))
	for i := 1; i <= numTrials; i++ {
		randomNumber := rd.Float64()
		if randomNumber < pHeads {
			heads++
		} else {
			tails++
		}
	}
	fmt.Println("In", numTrials, "trials there were", heads, "heads and", tails, "tails")
	fmt.Println("PERCENT HEADS:", float64(100)*float64(heads)/float64(numTrials), "percent")
}
